cmd/experimental/kjobctl/test/e2e/slurm_test.go

```diff
@@ -0,0 +1,315 @@
+			"with --first-node-ip",
+			[]string{"--first-node-ip"}, []string{"--array", "1-5%2", "--nodes", "2", "--ntasks", "2"},
+			int32(3), int32(2),
+			map[string]string{
+				"SLURM_NTASKS_PER_NODE":   "2",
+				"SLURM_ARRAY_JOB_ID":      "1",
+				"SLURM_MEM_PER_CPU":       "",
+				"SLURM_GPUS":              "",
+				"SLURM_NNODES":            "2",
+				"SLURM_MEM_PER_GPU":       "",
+				"SLURM_NTASKS":            "2",
+				"SLURM_ARRAY_TASK_COUNT":  "5",
+				"SLURM_TASKS_PER_NODE":    "2",
+				"SLURM_CPUS_PER_TASK":     "",
+				"SLURM_ARRAY_TASK_MAX":    "5",
+				"SLURM_CPUS_PER_GPU":      "",
+				"SLURM_SUBMIT_DIR":        "/slurm/scripts",
+				"SLURM_NPROCS":            "2",
+				"SLURM_CPUS_ON_NODE":      "",
+				"SLURM_ARRAY_TASK_MIN":    "1",
+				"SLURM_JOB_NODELIST":      "profile-slurm-xxxxx-0.profile-slurm-xxxxx,profile-slurm-xxxxx-1.profile-slurm-xxxxx",
+				"SLURM_JOB_CPUS_PER_NODE": "",
+				"SLURM_JOB_FIRST_NODE":    "profile-slurm-xxxxx-1.profile-slurm-xxxxx",
+				"SLURM_MEM_PER_NODE":      "",
+				"SLURM_JOB_FIRST_NODE_IP": "",
+			},
+			[]map[string]map[string]string{
+				{
+					"c1-0": {
+						"SLURM_ARRAY_TASK_ID": "1",
+						"SLURM_JOB_ID":        "1",
+						"SLURM_JOBID":         "1",
+						"SLURM_SUBMIT_HOST":   "profile-slurm-xxxxx-0",
+					},
+					"c1-1": {
+						"SLURM_ARRAY_TASK_ID": "2",
+						"SLURM_JOB_ID":        "2",
+						"SLURM_JOBID":         "2",
+						"SLURM_SUBMIT_HOST":   "profile-slurm-xxxxx-0",
+					},
+				},
+				{
+					"c1-0": {
+						"SLURM_ARRAY_TASK_ID": "3",
+						"SLURM_JOB_ID":        "3",
+						"SLURM_JOBID":         "3",
+						"SLURM_SUBMIT_HOST":   "profile-slurm-xxxxx-1",
+					},
+					"c1-1": {
+						"SLURM_ARRAY_TASK_ID": "4",
+						"SLURM_JOB_ID":        "4",
+						"SLURM_JOBID":         "4",
+						"SLURM_SUBMIT_HOST":   "profile-slurm-xxxxx-1",
+					},
+				},
+				{
+					"c1-0": {
+						"SLURM_ARRAY_TASK_ID": "5",
+						"SLURM_JOB_ID":        "5",
+						"SLURM_JOBID":         "5",
+						"SLURM_SUBMIT_HOST":   "profile-slurm-xxxxx-2",
+					},
+				},
+			},
+			true,
+		),
+	)
+})
+
+func parseSlurmCreateOutput(output []byte, profileName string) (string, string, string) {
+	output = bytes.ReplaceAll(output, []byte("\n"), []byte(""))
+	re := regexp.MustCompile(fmt.Sprintf(`^job\.batch\/(%[1]s-slurm-.{5}) createdconfigmap\/(%[1]s-slurm-.{5}) createdservice\/(%[1]s-slurm-.{5}) created$`, profileName))
+	matches := re.FindSubmatch(output)
+
+	var jobName, configMapName, serviceName string
+	if len(matches) > 1 {
+		jobName = string(matches[1])
+	}
+	if len(matches) > 2 {
+		configMapName = string(matches[2])
+	}
+	if len(matches) > 3 {
+		serviceName = string(matches[3])
+	}
+
+	return jobName, configMapName, serviceName
+}
+
+func parseSlurmEnvOutput(output []byte) map[string]string {
+	parts := bytes.Split(output, []byte{'\n'})
+	gotOut := make(map[string]string, len(parts))
+	for _, part := range parts {
+		pair := bytes.Split(part, []byte{'='})
```

Comment:
Do we want to make it safer for cases where `=` might become in the actual values? 
```suggestion
		pair := bytes.SplitN(part, []byte{'='}, 2)
```

Reply:
Done